src/context/filter.rs

```diff
@@ -0,0 +1,155 @@
+use crate::{
+    Context, RedisModuleCommandFilter, RedisModuleCommandFilterCtx, RedisModuleString,
+    RedisModule_CommandFilterArgDelete, RedisModule_CommandFilterArgGet,
+    RedisModule_CommandFilterArgInsert, RedisModule_CommandFilterArgReplace,
+    RedisModule_CommandFilterArgsCount, RedisModule_CommandFilterGetClientId,
+    RedisModule_CreateString, RedisModule_RegisterCommandFilter,
+    RedisModule_UnregisterCommandFilter, ValkeyString,
+};
+use std::ffi::{c_int, CString};
+use std::ptr::null_mut;
+use std::str::Utf8Error;
+
+#[derive(Debug, Clone, Copy)]
+pub struct CommandFilter {
+    inner: *mut RedisModuleCommandFilter,
+}
+
+// otherwise you get error:     cannot be shared between threads safely
+unsafe impl Send for CommandFilter {}
+unsafe impl Sync for CommandFilter {}
+
+/// CommandFilter is a wrapper around the RedisModuleCommandFilter pointer
+///
+/// It provides a way to check if the filter is null and to create a new CommandFilter
+impl CommandFilter {
+    pub fn new(inner: *mut RedisModuleCommandFilter) -> Self {
+        CommandFilter { inner }
+    }
+
+    pub fn is_null(&self) -> bool {
+        self.inner.is_null()
+    }
+}
+
+pub struct CommandFilterCtx {
+    inner: *mut RedisModuleCommandFilterCtx,
+}
+
+/// wrapping the RedisModuleCommandFilterCtx to provide a higher level interface to call RedisModule_CommandFilter*
+///
+/// provides functions to interact with the command filter context, such as getting the number of arguments, getting and replacing arguments, and deleting arguments.
+impl CommandFilterCtx {
+    pub fn new(inner: *mut RedisModuleCommandFilterCtx) -> Self {
+        CommandFilterCtx { inner }
+    }
+
+    /// wrapper for RedisModule_CommandFilterArgsCount
+    pub fn args_count(&self) -> c_int {
+        unsafe { RedisModule_CommandFilterArgsCount.unwrap()(self.inner) }
+    }
+
+    /// wrapper for RedisModule_CommandFilterArgGet
+    pub fn arg_get(&self, pos: c_int) -> *mut RedisModuleString {
+        unsafe { RedisModule_CommandFilterArgGet.unwrap()(self.inner, pos) }
+    }
+
+    /// wrapper to get argument as a Result<&str, Utf8Error> instead of RedisModuleString
+    pub fn arg_get_as_str<'a>(&self, pos: c_int) -> Result<&'a str, Utf8Error> {
+        let arg = self.arg_get(pos);
+        ValkeyString::from_ptr(arg)
+    }
+
+    /// wrapper to get 0 argument, the command which is always present and return as &str
+    pub fn cmd_get_as_str(&self) -> &str {
+        let cmd = self.arg_get(0);
+        ValkeyString::from_ptr(cmd).unwrap()
```

Comment:
Could we instead return the result itself by updating the function signature to `Result<&'a str, Utf8Error>`? We can let the module developers handle this error

Reply:
updated to `cmd_get_try_as_str`